Sources/BraintreeCore/BTPaymentMethodNonceParser.swift

```diff
@@ -53,18 +53,80 @@ import Foundation
             return nil
         }
 
-        if let completionHandler = completionHandler {
+        if let completionHandler {
             return completionHandler(json)
         }
 
-        if json?["nonce"].isString != false {
+        if json?["nonce"].isString == false {
+            return nil
+        }
+
+        let type = json?["type"].asString()
+
+        if type == "CreditCard", let cardType = json?["details"]["cardType"].asString() {
+            return BTPaymentMethodNonce(
+                nonce: json?["nonce"].asString() ?? "",
+                type: self.cardType(from: cardType),
+                isDefault: json?["default"].isTrue ?? false
+            )
+        } else if type == "ApplePayCard" {
+            return BTPaymentMethodNonce(
+                nonce: json?["nonce"].asString() ?? "",
+                type: json?["details"]["cardType"].asString() ?? "ApplePayCard",
+                isDefault: json?["default"].isTrue ?? false
+            )
+        } else if type == "PayPalAccount" {
+            return BTPaymentMethodNonce(
+                nonce: json?["nonce"].asString() ?? "",
+                type: "PayPal",
+                isDefault: json?["default"].isTrue ?? false
+            )
+        } else if type == "VenmoAccount" {
+            return BTPaymentMethodNonce(
+                nonce: json?["nonce"].asString() ?? "",
+                type: "Venmo",
```

Comment:
It looks like we previously had used `VenmoAccount` ([see V5 load method](https://github.com/braintree/braintree_ios/blob/75a48494b48dc83204ced72a0cee68057a8e7170/Sources/BraintreeVenmo/BTVenmoDriver.m#L49-L51))

Reply:
That is just for the register type piece but under the hood the type was passed as `Venmo` - see [this unit test](https://github.com/braintree/braintree_ios/blob/75a48494b48dc83204ced72a0cee68057a8e7170/UnitTests/BraintreeVenmoTests/BTPaymentMethodNonceParser_Venmo_Tests.swift#L20) in v5. I did test this in the demo app by calling this method to verify that the types would be `Venmo` and `PayPal` in v5.

Reply:
Got it, thanks for confirming Jax!